feat(nuxt): Improve Nuxt server-side setup docs (#12102)
getsentry/sentry-docs: docs/platforms/javascript/guides/nuxt/install/dynamic-import.mdx

## Understanding the `import()` expression

<Alert level='warning'>

>   This installation method does not work with more recent versions of Nuxt/Nitro.

Reviewer: ```suggestion
  This installation method doesn't work with more recent versions of Nuxt/Nitro.
```

>   This installation method comes with restrictions when using more recent versions of Nuxt/Nitro.

Reviewer 2: ```suggestion
  This installation method does not work with more recent versions of Nuxt/Nitro.
```

is this not more correct? 😅

Author: It can work sometimes 😄 I had it working with newer versions, but only under certain conditions.

But yeah, probably it's better to just say it's not working to be more precise.

>   We recommend the guide for installing the SDK with the <PlatformLink to="/install/cli-import">CLI flag `--import`</PlatformLink> or <PlatformLink to="/install/limited-server-tracing">limited server tracing</PlatformLink>

Reviewer: ```suggestion
  We recommend reading the guide for installing the SDK with the <PlatformLink to="/install/cli-import">CLI flag `--import`</PlatformLink> or <PlatformLink to="/install/limited-server-tracing">limited server tracing</PlatformLink>
```

>   We recommend the guide for installing the SDK with the <PlatformLink to="/install/cli-import">CLI flag `--import`</PlatformLink> or a <PlatformLink to="/install/top-level-import/">top-level import</PlatformLink>

Reviewer 2: ```suggestion
  We recommend the guide for installing the SDK with the <PlatformLink to="/install/cli-import">CLI flag `--import`</PlatformLink> or a <PlatformLink to="/install/limited-server-tracing/">limited server tracing</PlatformLink>
```

</Alert>

> The `import()` expression, or dynamic import, enables flexible, conditional module loading in ESM.
> For the Sentry Nuxt SDK, it provides an approach to initialize server-side Sentry configuration before application startup.

Reviewer: ```suggestion
The `import()` expression, or dynamic import, enables flexible, conditional module loading in ESM. In the Sentry Nuxt SDK, it makes it possible to set up server-side Sentry configuration before the application starts.
```

> The Sentry Nuxt SDK will be initialized before any other code runs and the Nitro server runtime code will be loaded later because of `import()`ing it.
> This early initialization is required to set up the SDK's instrumentation of various libraries correctly.

Reviewer: ```suggestion
The Sentry Nuxt SDK initializes before any other code runs, while the Nitro server runtime loads later because it is `import()`ed. This early setup ensures that the SDK instruments various libraries correctly.
```

}
```

You can also check out the guide for installing the SDK with the <PlatformLink to="/install/cli-import">CLI flag `--import`</PlatformLink> or <PlatformLink to="/install/limited-server-tracing">limited-server-tracing</PlatformLink>.

## Initializing Sentry with Dynamic `import()`

Enable the dynamic `import()` by setting `autoInjectServerSentry`:

```typescript {filename:nuxt.config.ts} {3}
export default defineNuxtConfig({
  sentry: {
    autoInjectServerSentry: 'experimental_dynamic-import'
  },
})
```

> After setting this, the Sentry Nuxt SDK adds some build-time configuration to ensure your app is wrapped with `import()`.
> With this, Sentry can be initialized before all other modules of the application.

Reviewer: ```suggestion
After setting this, the Sentry Nuxt SDK will add build-time configuration so that your app will be wrapped with `import()`, ensuring that Sentry can be initialized before any other modules.
```

}
```

> You can also check out the guide for installing the SDK with the <PlatformLink to="/install/cli-import">CLI flag `--import`</PlatformLink> or a <PlatformLink to="/install/top-level-import/">top-level import</PlatformLink>.

Reviewer 2: ```suggestion
You can also check out the guide for installing the SDK with the <PlatformLink to="/install/cli-import">CLI flag `--import`</PlatformLink> or a <PlatformLink to="/install/limited-server-tracing/">limited-server-tracing</PlatformLink>.
```
